### ISSUE TITLE
GT6 food does not rot under EnviroMine system

### ISSUE THREAD

**Issue by xbony2** [GregTech6/gregtech6#94]
EnviroMine adds a rotting mechanic to food, which works for most food (vanilla, Pam's, etc...). However, GregTech 6 food does not rot for whatever reason.

**Comment by GregoriusT** [comment]
Uh it should work, are you sure it isn't just disabled in your Config? Either that or it requires AppleCore to work.

**Comment by xbony2** [comment]
This is using AppleCore 3.1.1, GregTech 6.15.01, EnviroMine 1.3.124, [among other mods](https://www.curseforge.com/minecraft/modpacks/wikipack-5). Food spoiling is clearly on in the EnviroMine config (and works for non-GT6 food in-game). I don't see anything in `gregtech.cfg` or other GT6 configs. Am I missing something?

**Comment by GregoriusT** [comment]
I guess it is a problem because I dont use the same System for my Food that vanilla MC uses. So I think I might have confused rotting Foods in the Betweenlands Dimension with this mechanic. XD

**Comment by xbony2** [comment]
Oh, I forgot that the Betweenlands has some weird mechanics. But yeah this is different. I don't know if it would be an easy thing to fix but it's a bit of an exploit.

**Comment by tyrael93** [comment]
I think this might be related to how GT6 Food Items are different from regular ones, specifically because they use Metadata and aren’t extending ItemFood. This also happens with custom backport of nutrition which doesn't register gregfood as having nutritional value and that might also be the reason why GT food doesn't rot following the TFC rules, so it makes sense that it wouldn't rot with enviromine as well. Basically if I understand it correctly greg's food doesn't play nice with most mods because they're working on a different base.

**Comment by GregoriusT** [comment]
Yep, that is correct, though in case of Apple Core , if there is an API i do use it.

**Comment by tyrael93** [comment]
By accident I was going around the enviromine repo and I found someone opened an issue about this same problem 7 years ago called [Gregtech food will not Rot?](https://github.com/EnviroMine/EnviroMine-1.7/issues/83 )
Given the timeframe, I think this can definitely be closed :laughing:

**Comment by xbony2** [comment]
Well, it's still an issue. I am hoping it something that Greg could fix on his own. Issue mentions something about a custom spoiling config so I could look into that on my end (although I have to take care of some stuff currently so can't).

**Comment by tyrael93** [comment]
Nah, he had no idea what he was saying and just closed the issue without looking into what was happening under the hood like a doofus, thinking that it was the user's fault :roll_eyes: 
The only real reply here is greg's, gregfood literally cannot rot because enviromine doesn't even realize it's food, and it would be impossible to check for it without reworking the mod around it as a special case. I just found it funny to see the same issue mentioned 7 years ago :rofl:

**Comment by NeveHanter** [comment]
Just for future reference, on my side adding something like that:
```
spoiling {

    item_gt_multiitem_food {
        S:01.ID=gregtech:gt.multiitem.food
        I:02.Damage=-1

        # Set blank to rot into nothing
        S:"03.Rotten ID"=enviromine:rottenFood
        I:"04.Rotten Damage"=0

        # Set this to -1 to disable rotting on this item
        I:"05.Days To Rot"=7
    }
}
```
to `.minecraft/config/enviromine/profiles/default/CustomProperties/gregtech.cfg` added spoiling to GT food.

The only important note here is that you need to transfer the items between the container and your inventory for it to refresh, but maybe it will refresh itself after some time, I didn't check that.

**Comment by tyrael93** [comment]
The issue with a blanket ban like that is that will also rot drinks, bottles, liquors, cans, etc etc etc, for hundreds of items since greg extends the multiitem.food ID for god knows how many things. I think a bunch of those are not even actual edible items.

That said, the whole way enviromine food rot is enabled is incredibly clunky because every item gets its own NBT tag: pick up 16 apples from cutting a tree, they now all occupy 16 different spots in a chest.
TFC has a similar implementation, but the fix is simple, just cut the rot to refresh the NBT values and stack the food. Also, if after the arbitrary week that blackberry is going to rot in say, an hour, you can just juice it and put it in a bottle and it will be good for another arbitrary week. Same applies to all kinds of juices by the way, so just put the fluid in a pot or a jug and pick it up again to refresh it. For a similar loophole with enviromine and vanilla food, you can turn wheat into haybales, which again, don't rot, and then unpack them to get free fresh wheat.

So as a whole, it's a broken system that cannot be fixed. GT6 just exacerbates the issue and strains these pre-existing problems.

**Comment by GregoriusT** [comment]
I am not sure if literally every metadata on that Item should rot in the first place, this clearly is a wildcard thing. But if that is acceptable to you, sure. I dont think its good enough for my standards though

**Comment by tyrael93** [comment]
> I am not sure if literally every metadata on that Item should rot in the first place, this clearly is a wildcard thing. But if that is acceptable to you, sure. I dont think its good enough for my standards though

Can you technically specify the metadata for each entry by editing        `I:02.Damage=-1` ,you think?
That would make it a lot more granular, if someone had the patience of hand-picking every single food item in the config.
